test/e2e_node/services/kubelet.go

```diff
@@ -224,6 +257,94 @@ func (e *E2EServices) startKubelet() (*server, error) {
 	return server, server.start()
 }
 
+// splitKubeletConfigArgs parses args onto a KubeletConfiguration object and also returns the unknown args
+func splitKubeletConfigArgs(args []string) (*kubeletconfig.KubeletConfiguration, []string, error) {
+	kc, err := options.NewKubeletConfiguration()
+	if err != nil {
+		return nil, nil, err
+	}
+	fs := pflag.NewFlagSet("kubeletconfig", pflag.ContinueOnError)
+	options.AddKubeletConfigFlags(fs, kc)
+	known, other := splitKnownArgs(fs, args)
+	if err := fs.Parse(known); err != nil {
+		return nil, nil, err
+	}
+	return kc, other, nil
+}
+
+// splitKnownArgs splits argument list into those known by the flagset, and those not known
+// only tests for longhand args, e.g. prefixed with `--`
+// TODO(mtaufen): I don't think the kubelet has any shorthand args, but if it does we will need to modify this.
+func splitKnownArgs(fs *pflag.FlagSet, args []string) ([]string, []string) {
+	known := append([]string(nil), args...)
+	other := []string{}
+	lastFlag := len(known)
+	for i := len(known) - 1; i >= 0; i-- {
```

Comment:
I think it's more clear to create 2 new array, `known` and `other`.
And append to each of them correspondingly.

This is too confusing and error prone.

Reply:
I updated it to make the construction behavior more clear.
I think it's easier to walk backwards, rather than look ahead for the next flag occurrence.